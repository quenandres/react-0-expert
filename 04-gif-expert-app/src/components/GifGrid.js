import React from 'react'

export const GifGrid = ( { category } ) => {
    
    const getGifs = async () => {
        const url = 'https://api.giphy.com/v1/gifs/search?q=Rick and morty&limit=10&api_key=xJw4xi7Ze2WIz1WMAEkpY9Ppnc7NUkKF';
        const resp = await fetch( url );
        const { data } = await resp.json();
        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images.downsized_medium.url
            }
        })
        console.log(gifs);
    }

    getGifs();

    return (
        <div>
            <h3>{category}</h3>
        </div>
    )
}
