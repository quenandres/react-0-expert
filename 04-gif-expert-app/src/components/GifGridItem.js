import React from 'react'
import '../index.css';
import PropTypes from 'prop-types';

export const GifGridItem = ( {title, url} ) => {

    return (
        <div className="card animate__rubberBand">
            <h1>Test</h1>
            <img src={ url } alt={ title } />
            <p> { title } </p>
        </div>
    )
}

GifGridItem.prototypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
