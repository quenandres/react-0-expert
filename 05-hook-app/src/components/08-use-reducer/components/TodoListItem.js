import React from 'react'

export const TodoListItem = ({key, todo, handleToggle, handleDelete}) => {
    return (
        <li 
            className='list-group-item'
            key={ todo.id }
        >
            <p 
                className={`${ todo.done && 'complete' }`}
                onClick={ () => handleToggle(todo.id)}
            > { key + 1 }. { todo.description } </p>
            <button 
                className='btn btn-danger'
                onClick={ () => handleDelete(todo.id) }
            >Borrar</button>
        </li>
    )
}
