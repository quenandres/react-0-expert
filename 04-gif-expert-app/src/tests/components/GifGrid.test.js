import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')

describe('test para <GifGrid />', () => {

    const category = 'One punch';
    const mockUseFetchGifs = useFetchGifs;
    const gifs = [{
        id: 'ABC',
        url: 'https://localhost/something',
        title: 'Lorem ipsum'
    }];
    beforeEach(() => {
        useFetchGifs.mockImplementation( category => {
            return { data: gifs, loading: false}
          });
    });

    test('debe mostrarse correctamente', () => {
        //Cuando se llame el mock
        mockUseFetchGifs.mockImplementation( category => {
            return { data: [], loading: true }
        });
        
        const wrapper = shallow(<GifGrid category = {category}/>);
        const title = wrapper.find('h3').prop('children');
        expect(wrapper).toMatchSnapshot();
        expect(category).toBe(title);
    });

    test('debe mostrar items cuando cargan imagenes useFetchGifs', () => {
        const data = [{
          id: '0',
          url: 'https://lorem.com/img',
          title: 'mockTitle'
        }];
          //Fingir un mock        
        mockUseFetchGifs.mockImplementation( category => {
          return { data, loading: false };
        });
        const wrapper = shallow( <GifGrid category={ category } /> );
        expect( wrapper ).toMatchSnapshot();

    });
})
