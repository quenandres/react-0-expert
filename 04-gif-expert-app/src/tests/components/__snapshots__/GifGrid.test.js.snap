// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test para <GifGrid /> debe mostrar items cuando cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3>
    One punch
  </h3>
  <div
    className="card-grid animate__animated animate__swing"
  >
    <GifGridItem
      id="123"
      key="123"
      title="mockTitle"
      url="https://lorem.com/img"
    />
  </div>
</Fragment>
`;

exports[`test para <GifGrid /> debe mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    One punch
  </h3>
  <p
    className="animate__animated animate__bounceInRight"
  >
    loading...
  </p>
  <div
    className="card-grid animate__animated animate__swing"
  />
</Fragment>
`;
